h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java

```diff
@@ -828,6 +828,248 @@ private Vec createRandomBinaryWeightsVec(final long len, final int randomSeed) {
+
+          assertEquals(originalCrossValidationMetrics.rmse(), crossValidationModelMetrics.rmse(), 1e-20);
+          assertEquals(originalCrossValidationMetrics._nobs, crossValidationModelMetrics._nobs, 1e-20);
+          assertEquals(originalCrossValidationMetrics._sigma, crossValidationModelMetrics._sigma, 1e-20);
+          assertEquals(originalCrossValidationMetrics._logloss, crossValidationModelMetrics._logloss, 1e-20);
+          assertEquals(originalCrossValidationMetrics._auc._auc, crossValidationModelMetrics._auc._auc, 1e-20);
+          assertArrayEquals(originalCrossValidationMetrics._domain, crossValidationModelMetrics._domain);
+        } finally {
+          if (crossValidationModel != null) {
+            crossValidationModel.deleteCrossValidationPreds();
+            crossValidationModel.deleteCrossValidationModels();
+            crossValidationModel.remove();
+          }
+        }
+
+      }
+
+    } finally {
+      Scope.exit();
+      if (airlinesFrame != null) airlinesFrame.remove();
+      if (model != null) {
+        model.deleteCrossValidationModels();
+        model.deleteCrossValidationPreds();
+        model.remove();
+      }
+      if (crossValidationSubset != null) {
+        crossValidationSubset.remove();
+      }
+    }
+
+  }
+
+  @Test
+  public void testRegressionCrossValidationWeights() {
+    Frame prostateFrame = null;
+    Frame crossValidationSubset = null;
+    XGBoostModel model = null;
+    Scope.enter();
+    try {
+      prostateFrame = parse_test_file("./smalldata/prostate/prostate.csv");
+      Scope.track(prostateFrame.replace(8, prostateFrame.vecs()[8].toCategoricalVec()));   // Convert GLEASON to categorical
+      DKV.put(prostateFrame);
+
+      final Vec foldColumnVector = createRandomFoldColumn(prostateFrame.numRows(), 2, 10);
+      final String foldColumnName = "foldColumn";
+      prostateFrame.add(foldColumnName, foldColumnVector);
+
+      XGBoostModel.XGBoostParameters parms = new XGBoostModel.XGBoostParameters();
+      parms._dmatrix_type = XGBoostModel.XGBoostParameters.DMatrixType.auto;
+      parms._response_column = "AGE";
+      parms._train = prostateFrame._key;
+      parms._ignored_columns = new String[]{"ID"};
+      parms._fold_column = foldColumnName;
+
+      model = new hex.tree.xgboost.XGBoost(parms).trainModel().get();
+      assertEquals(ModelCategory.Regression, model._output.getModelCategory());
+      assertEquals(foldColumnName, model._output.foldName());
+      parms._fold_column = null; // No longer required
+
+      Key[] originalCrossValidationModels = model._output._cross_validation_models;
+      assertEquals(2, originalCrossValidationModels.length);
+
+      for (int i = 0; i < originalCrossValidationModels.length; i++) {
+
+        crossValidationSubset = Rapids.exec(String.format("(rows %s ( == (cols %s [9]) %d))", prostateFrame._key, prostateFrame._key, originalCrossValidationModels.length - (i + 1))).getFrame();
```

Comment:
each subset should have ideally a unique key and should be deleted explicitly

Reply:
It is already deleted explicitely in the finally block. The key is now however generated randomly (that is what I changed).